pkg/operator/k8sutil/pod.go

```diff
@@ -89,19 +90,34 @@ func NodeEnvVar() v1.EnvVar {
 }
 
 // ConfigDirEnvVar config dir env var
-func ConfigDirEnvVar() v1.EnvVar {
-	return v1.EnvVar{Name: "ROOK_CONFIG_DIR", Value: DataDir}
+func ConfigDirEnvVar(dataDir string) v1.EnvVar {
+	return v1.EnvVar{Name: "ROOK_CONFIG_DIR", Value: dataDir}
 }
 
 func GetContainerImage(pod *v1.Pod, name string) (string, error) {
+	return GetSpecContainerImage(pod.Spec, name)
+}
 
-	image, err := GetMatchingContainer(pod.Spec.Containers, name)
+func GetSpecContainerImage(spec v1.PodSpec, name string) (string, error) {
+	image, err := GetMatchingContainer(spec.Containers, name)
 	if err != nil {
 		return "", err
 	}
 	return image.Image, nil
 }
 
+func GetContainerVersion(spec v1.PodSpec, name string) (string, error) {
+	image, err := GetSpecContainerImage(spec, name)
+	if err != nil {
+		return "", err
+	}
+	parts := strings.Split(image, ":")
```

Comment:
This maybe overkill but isn't there a library exactly for parsing/"cleaning" container image names?

Reply:
actually i didn't end up needing the image version, so will remove it. If needed in the future, i'll look for the library, i just don't recall seeing it before.